Makefile

```diff
@@ -55,5 +58,9 @@ clean:  ## Remove all build files.
 		htmlcov/ \
 		tmp/
 
+install-git-hooks:  ## Install GIT pre-commit hook.
+	ln -sf ../../.git-pre-commit .git/hooks/pre-commit
+	chmod +x .git/hooks/pre-commit
+
```

Comment:
You should be able to optimize this `Makefile` a bit by doing something like this:

```make
.git/hooks/pre-commit:
	ln -sf ../../.git-pre-commit .git/hooks/pre-commit
	chmod +x .git/hooks/pre-commit

...

test: .git/hooks/pre-commit
        ...
```

The make will conditionally install the githook if it is missing (won't run the target every time another target is run).

Reply:
In `test` target I already do `${MAKE} install-git-hooks`; isn't it the same?

Reply:
Close, but the path to the symlink acts as an artifact, so `make` will do nothing if the link already exists. Using a named target, `make` does not know about the artifact, and will symlink every time you invoke the target.